worker/src/main/scala/org/apache/celeborn/service/deploy/worker/FetchHandler.scala

```diff
@@ -223,8 +227,8 @@ class FetchHandler(
           // 1. when the current request is a non-range openStream, but the original unsorted file
           //    has been deleted by another range's openStream request.
           // 2. when the current request is a range openStream request.
-          if ((endIndex != Int.MaxValue) || (endIndex == Int.MaxValue && !fileInfo.addStream(
-              streamId))) {
+          if (shuffleNeedSort && ((endIndex != Int.MaxValue) ||
+              (endIndex == Int.MaxValue && !fileInfo.addStream(streamId)))) {
```

Comment:
nit: To improve clarity a bit
```suggestion
          if (shuffleNeedSort &&
                (endIndex != Int.MaxValue || (endIndex == Int.MaxValue && !fileInfo.addStream(streamId)))) {
```

Reply:
Done